app/models/observation.rb

```diff
@@ -193,6 +193,50 @@ class Observation < AbstractModel
   # Automatically (but silently) log destruction.
   self.autolog_events = [:destroyed]
 
+  include CreatedUpdatedScopes
+
+  scope :include_all_name_proposals, lambda { |name|
+    joins(:namings).where(namings: { name: name })
+  }
+  scope :of_look_alikes, lambda { |name|
+    joins(:namings).where(namings: { name: name }).
+      where(Observation[:name] != name)
+  }
+  scope :include_subtaxa, lambda { |text_name|
+    names = {}
+    starting_names = Name.text_name_like(text_name).with_correct_spelling
+    starting_names.each do |name|
+      names = starting_names.or(Name.related_taxa(name)).distinct
+    end
+    where(name: names)
+  }
+  scope :of_synonyms, ->(name) { where(name: name.synonyms) }
+  scope :of_name, ->(name) { where(name: name) }
```

Comment:
I've been trying to follow the flurry of emails surrounding this section, but couldn't understand.  I'm afraid I don't like this solution.  I see these as modifiers of a single "of_name" scope.  Is it possible to write a single scope with named parameters instead?  I think that would be closer to the original implementation and intention.  And I think it would be easier to understand.  Although, as usual, making wordier names of what you have could work in a pinch.

For example, Observation.include_subtaxa("Agaricus") sounds really strange to me.  It should be Observation.of_subtaxa_of_name.  Wait, no, it should be Observation.of_name_or_subtaxa?  It's not clear as written thanks to the strangeness of the existing naming of "text_name_like" and "name_like", I forget which gives subtaxa and which just checks if the name contains the given string... neither actually full sense to me in this case, though.

What I'm recommending instead would be Observation.of_name("Agaricus", include_subtaxa: true, include_misspellings: false, include_all_name_proposals: false, etc.)  I think this is very clear (in intent), very flexible, and if I remember right, much closer to the original implementation and intention.  Yes, it makes this scope's lambda huge.  But it can be split up internally with private helpers, etc.

Obviously, as always, ad nauseum, you all are totally free to overrule me, but you did ask for my opinion...

Reply:
Right! This is complex to write, but also clearer in that the arguments add `or` conditions on the original scope `of_name` — they are not scope-defining on their own. Okay so it might be like

```ruby
scope :of_name, lambda {
  |name, include_subtaxa?, include_misspellings?, include_all_name_proposals?, 
  include_synonyms?, include_lookalikes?, include_whatever?|
  scope = where(name: name)
  scope = if include_subtaxa
                scope.or(condition)
              else
                scope
              end
  # etc.
  scope
}
```

Reply:
Should I name those lambda args with question marks to imply they are `booleanisms?` 

Also if you have any coding patterns to suggest...
I'm not great at writing trainyard switches like this, and I'm already wincing from the ABC cop's impending bludgeon

Reply:
I wrote one that seems ok, but i'm not sure I covered all the possible params. Also, I need to write tests for it!

Reply:
Ah, it goes much better when I splat the **args.
If you're checking it out, be sure you have the very recent commits; it's working pretty well now. 

FWIW I use `rails console --sandbox` to try calling the scope.to_sql, like
```ruby
> name = Name.last
# => <Name:0x00005567b3df0780>

> Observation.of_name(name, include_all_name_proposals: true).to_sql
# => "SELECT `observations`.* FROM `observations` INNER JOIN `namings` ON `namings`.`observation_id` = `observations`.`id` WHERE `namings`.`name_id` = 111323"
```